lib/SILOptimizer/Mandatory/SendNonSendable.cpp

```diff
@@ -633,6 +666,352 @@ class BlockPartitionState {
   }
 };
 
+enum class LocalConsumedReasonKind {
+  LocalConsumeInst,
+  LocalNonConsumeInst,
+  NonLocal
+};
+
+// Why was a value consumed, without looking across blocks?
+// kind == LocalConsumeInst: a consume instruction in this block
+// kind == LocalNonConsumeInst: an instruction besides a consume instruction
+//                              in this block
+// kind == NonLocal: an instruction outside this block
```

Comment:
good to move these explainer comments directly onto the `LocalConsumedReasonKind` cases instead of on top of the struct containing it so it shows up in IDE documentation windows for what `LocalConsumedReasonKind::X` means.